src/js/utils/PodUtil.js

```diff
@@ -36,7 +39,53 @@ var PodUtil = {
     return instance.getContainers().some(function (container) {
       return PodUtil.isContainerMatchingText(container, text);
     });
+  },
+
+  /**
+   * This function merges the output of MesosStateUtil.getPodHistoricalInstances
+   * with the array of currently known instances returned from the
+   * Pod.getInstances() function.
+   *
+   * NOTE: podInstances is an array of PodInstance objects, but the
+   *       historicalInstances is an array of plain objects, with a structure
+   *       that can be used to construct a PodInstance!
+   *
+   * @param {PodInstanceList} podInstances - An array of PodInstance objects
+   * @param {Array} historicalInstances - The output of getPodHistoricalInstances
+   * @returns {PodInstanceList} The new array of PodInstance objects
+   */
+  mergeHistoricalInstanceList(podInstances, historicalInstances) {
+    if (!historicalInstances) {
+      return podInstances;
+    }
+
+    // De-compose PodInstances into plain objects, so we always operate
+    // with plain objects
+    let podInstancesMap = podInstances.reduceItems(function (memo, instance) {
+      memo[instance.getId()] = instance.get();
+      return memo;
+    }, {});
+
+    historicalInstances.forEach(function (instance) {
```

Comment:
Every time I see `forEach` I think how it could be replaced with `map` or `reduce`